src/main/resources/avro/metadata.avdl

```diff
@@ -122,10 +122,25 @@ For data accessors, they're a simple way to scope exploration and analysis
 what's the distribution of that result in the data from our project?)   
 */
 record Dataset {
-  /** The dataset UUID. This is globally unique. */
+  /**
+  The dataset's id. This is the local/legacy/generic id.
+  */
   string id;
 
-  /** A description of the dataset. */
+  /**
+  The dataset's guid. This is globally unique id (UUIDv4).
+  TODO: Required?!
+  */
```

Comment:
I think an overall strategy for assigning guids to objects should be devised, which looks at _all_ the types before we add it here. I would favour removing this for now, and explicitly tackling GUIDs across the board. I don't really see any advantage to the data set having a GUID if ReadGroupSets and VariantSets don't.